api/v1alpha3/virtualmachine_types.go

```diff
@@ -242,8 +242,69 @@ type VirtualMachineImageRef struct {
 	Name string `json:"name"`
 }
 
+// VirtualMachineCdromSpec describes the desired state of a CD-ROM device.
+type VirtualMachineCdromSpec struct {
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Pattern="^[a-z0-9]{2,}$"
+
+	// Name describes the unique and immutable name of this CD-ROM device.
+	// If omitted or empty, the controller will generate it as "cdrom" + "num",
+	// where "num" is the device's index in the spec.cdrom list.
+	Name string `json:"name"`
+
+	// +kubebuilder:validation:Required
+
+	// Image describes the reference to an ISO type VirtualMachineImage or
+	// ClusterVirtualMachineImage resource used as the backing for the CD-ROM.
+	// If the image kind is omitted, it defaults to VirtualMachineImage.
+	//
+	// This field is immutable when the VM is powered on.
+	//
+	// Please note, unlike the spec.imageName field, the value of this
+	// spec.cdrom.image.name MUST be a Kubernetes object name.
+	Image *VirtualMachineImageRef `json:"image"`
+
+	// +optional
+	// +kubebuilder:default=true
+
+	// Connect describes the desired connection state of the CD-ROM device.
+	//
+	// When omitted or true, the CD-ROM device is added and connected to the
+	// VM. If the device already exists, it's updated to a connected state.
+	//
+	// When explicitly set to false, the CD-ROM device is added but remains
+	// disconnected from the VM. If the CD-ROM device already exists, it's
+	// updated to a disconnected state.
+	Connect bool `json:"connect"`
```

Comment:
CC @aruneshpa

Reply:
So this was named "Disconnected" initially and later changed to "Connected" after discussing with the team internally last week. The main concern was that "Disconnected" could be confusing in the code as it's negative and especially when used in a conditional statement (e.g. `if !cdrom.disconnected` to check if it's connected).

I don't have a strong preference between them but might still lean towards "disconnected" so that it makes more sense in the API. As API fields are more user-facing, I think we should prioritize choosing a more user-friendly term than an implementation detail inside the controller reconciliation process. 

@aruneshpa If you don't feel strongly about this, I can change this back to "disconnected" (or open to discussing any other name if you have other thoughts/concerns).

Reply:
Updated this field back to `Disconnected`: https://github.com/vmware-tanzu/vm-operator/pull/594/files#diff-3d050153d168a9ae5555513d460312ace08ab8ed5cc51a751fb7bd9375e4ba6cR264-R272